src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewCompiler.cs

```diff
@@ -321,8 +332,9 @@ private ViewCompilerWorkItem CreateRuntimeCompilationWorkItem(string normalizedP
 
         protected virtual CompiledViewDescriptor CompileAndEmit(string relativePath)
         {
-            var codeDocument = _templateEngine.CreateCodeDocument(relativePath);
-            var cSharpDocument = _templateEngine.GenerateCode(codeDocument);
+            var projectItem = _projectEngine.FileSystem.GetItem(relativePath);
```

Comment:
Should we check that the item exists somewhere?

Reply:
Nope, we don't care if it exists; we're just trying to add file watchers for the ones that have a path. This enables `_ViewImports.cshtml` changing, being created/removed etc.